x-pack/plugins/apm/public/components/shared/FilterBar/__test__/DatePicker.test.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mount } from 'enzyme';
+import React from 'react';
+import { Provider } from 'react-redux';
+import { MemoryRouter } from 'react-router-dom';
+// @ts-ignore
+import configureStore from 'x-pack/plugins/apm/public/store/config/configureStore';
+import { DatePicker } from '../DatePicker';
+
+function mountPicker(search?: string) {
+  const store = configureStore();
+  let path = '/whatever';
+  if (search) {
+    path += `?${search}`;
+  }
+  const mounted = mount(
+    <Provider store={store}>
+      <MemoryRouter initialEntries={[path]}>
+        <DatePicker />
+      </MemoryRouter>
+    </Provider>
+  );
+  return { mounted, store };
+}
+
+describe('DatePicker', () => {
+  describe('date calculations', () => {
+    const fakeNow = new Date('2019-02-15T12:00:00.000Z');
+    const realDateNow = global.Date.now.bind(global.Date);
+
+    beforeAll(() => {
+      global.Date.now = jest.fn(() => fakeNow);
+    });
+
+    afterAll(() => {
+      global.Date.now = realDateNow;
+    });
```

Comment:
I think this reads pretty well. I noticed that it was failing on CI with this error so digged a bit into it:

```
plugins/apm/public/components/shared/FilterBar/__test__/DatePicker.test.tsx:37:7 
Type 'Date' is not assignable to type 'number'.

global.Date.now = jest.fn(() => fakeNow);
```

Another approach is to use `spyOn` - that way you can leave the cleanup to jest:
```ts
beforeAll(() => {
  const fakeNow = new Date('2019-02-15T12:00:00.000Z').getTime();
  jest.spyOn(global.Date, 'now').mockReturnValue(fakeNow);
});
```

You could even put it in testHelpers if we need it elsewhere:
```ts
function mockDate(date: string) {
  const fakeNow = new Date(date).getTime();
  jest.spyOn(global.Date, 'now').mockReturnValue(fakeNow);
}
```

Reply:
Yeah this way of mocking time isn't foolproof. I don't know if using jest makes anything better here, though. You still have to use `mockRestore()` or one of the other clearing methods to reset the spy, I think?

Reply:
The bug there was that I was using the Date object for the result of `Date.now` instead of calling `getTime()` to produce the number.

Reply:
> The bug there was that I was using the Date object for the result of Date.now instead of calling getTime() to produce the number.

Yes, I just thought it looked nicer if we could avoid doing the cleanup manually. I seem to recall jasmine's implementation did this but perhaps jest went away from that.